### PR TITLE
refactor(tests): integration tests now create a cluster per test (shared across features)

### PR DESCRIPTION
**Pull request by pmalek-sumo**
##### Description

Make it so that each test has a separate cluster that can be shared with different features within that test.

Each test will get a values file for helm installation from `values` directory that is based on the test name (the details of name generation can be found in `strings.ValueFileFromT()`)

The reason for this is to limit the amount of boilerplate in tests themselves but we cannot attach/map the values.yaml to the test itself so we do this mapping instead.

### CODE REVIEW

**Review by swiatekm-sumo — CHANGES_REQUESTED** [review]
Seems like a good middle-ground for what we need. Can you document how this setup works, and especially the magic values.yaml naming, in the README?

File: tests/integration/internal/types.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import (
+	"context"
+	"testing"
+
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+type StepFunc = func(context.Context, *testing.T, *envconf.Config) context.Context
+
+type TestEnvFunc = func(context.Context, *envconf.Config, *testing.T) (context.Context, error)
```
[the remark this answers is not in the document]
Comment (pmalek-sumo) [reply tests/integration/internal/types.go:12]: I don't believe that should be the case:

* `TestEnvFunc` "aliases" https://github.com/kubernetes-sigs/e2e-framework/blob/ea87ca9689e6bcd00a0230991296f025baa27ec6/pkg/internal/types/types.go#L38-L42
* `StepFunc` "aliases"  https://github.com/kubernetes-sigs/e2e-framework/blob/ea87ca9689e6bcd00a0230991296f025baa27ec6/pkg/features/feature.go#L28

but actully the type for `StepFunc` is not internal so I just used what's defined in e2e module.

File: tests/integration/helm_otc_metadata_installation_test.go [hunk 2]
```diff
@@ -88,14 +83,24 @@ func Test_Helm_OT_Metadata(t *testing.T) {
...
 				assert.Eventually(t, func() bool {
-					var pvcs corev1.PersistentVolumeClaimList
-					err := envConf.Client().
-						Resources(namespace).
-						List(ctx, &pvcs,
-							resources.WithLabelSelector(fmt.Sprintf("app=%s-sumologic-otelcol-metrics", releaseName)),
-						)
-					return err == nil && len(pvcs.Items) == 3
+					t.Logf("kubeconfig: %s", ctxopts.KubectlOptions(ctx).ConfigPath)
+					cl, err := k8s.GetKubernetesClientFromOptionsE(t, ctxopts.KubectlOptions(ctx))
```
[the remark this answers is not in the document]
Comment (pmalek-sumo) [reply tests/integration/helm_otc_metadata_installation_test.go]: Good point!

[the remark this answers is not in the document]
Comment (pmalek-sumo) [reply tests/integration/helm_otc_metadata_installation_test.go]: Changed

**Comment by pmalek-sumo** [comment]
@swiatekm-sumo Added some info in the README about the test name <-> values file mapping.

PTAL

File: tests/integration/internal/types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (swiatekm-sumo) [reply tests/integration/internal/types.go]: So it'a e2e framework being inconsistent? Oh well.

File: tests/integration/README.md [hunk 3]
```diff
@@ -41,6 +41,20 @@ This will:
 - run the tests (using Go toolchain i.e. `go test ....`)
 - destroy created `kind` cluster(s)
 
+> **NOTE**: There is an assumption about mapping tests to helm values files.
+>
+> In order to keep the model of "one `kind` cluster per one test" we map the test name
+> (details can be find in [`strings.ValueFileFromT()`](./internal/strings/strings.go)) )
```
Comment (swiatekm-sumo) [inline tests/integration/README.md:47]: find -> found

Comment (pmalek-sumo) [reply tests/integration/README.md:47]:
```suggestion
> (details can be found in [`strings.ValueFileFromT()`](./internal/strings/strings.go)) )
```

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**